src/mds/CDir.h

```diff
@@ -782,6 +590,181 @@ class CDir : public MDSCacheObject, public Counter<CDir> {
   void print(ostream& out) override;
   void dump(Formatter *f, int flags = DUMP_DEFAULT) const;
   void dump_load(Formatter *f);
+
+  // context
+  MDCache *cache;
+
+  CInode *inode;  // my inode
+  frag_t frag;   // my frag
+
+  fnode_t fnode;
+  snapid_t first;
+  mempool::mds_co::compact_map<snapid_t,old_rstat_t> dirty_old_rstat;  // [value.first,key]
+
+  // my inodes with dirty rstat data
+  elist<CInode*> dirty_rstat_inodes;
+
+  version_t projected_version;
```

Comment:
To fix the warning, I have made projected_version public.

Reply:
@varshar16 Please see the definition of `-Wreorder`. It comes because ` The variables are listed within the class in a different order than in the initializer list.` 

The following is the fix for it, keeping `version_t projected_version;` at L668 in CDir.h:
```
jcollin@senta04:~/ceph/build/src/mds$ git diff ../../../src/mds/CDir.cc
diff --git a/src/mds/CDir.cc b/src/mds/CDir.cc
index 687021d..71c1d9d 100644
--- a/src/mds/CDir.cc
+++ b/src/mds/CDir.cc
@@ -194,9 +194,9 @@ CDir::CDir(CInode *in, frag_t fg, MDCache *mdcache, bool auth) :
   cache(mdcache), inode(in), frag(fg),
   first(2),
   dirty_rstat_inodes(member_offset(CInode, dirty_rstat_item)),
-  projected_version(0),
   dirty_dentries(member_offset(CDentry, item_dir_dirty)),
   item_dirty(this), item_new(this),
+  projected_version(0),
   num_head_items(0), num_head_null(0),
   num_snap_items(0), num_snap_null(0),
   num_dirty(0), committing_version(0), committed_version(0),

```
In addition, Do not change the scope of a private/protected variable. Write public functions instead, if necessary.

Reply:
Thanks @joscollin for the clarification, I will update the patch.